packages/viewer/src/components/scene-tree/lib/controller.ts

```diff
@@ -989,7 +991,20 @@ export class SceneTreeController {
     req: (metadata: grpc.Metadata) => ResponseStream<R>
   ): ResponseStream<R> {
     const metadata = this.createJwtMetadata(jwt);
-    return req(metadata);
+    return this.connectOptions.subscriptionUseWebSocketTransport
+      ? this.serverStreamWithWebSocket(() => req(metadata))
+      : req(metadata);
+  }
+
+  private serverStreamWithWebSocket<R>(
+    req: () => ResponseStream<R>
+  ): ResponseStream<R> {
+    grpc.setDefaultTransport(grpc.WebsocketTransport());
+    const stream = req();
+    grpc.setDefaultTransport(
+      grpc.CrossBrowserHttpTransport({ withCredentials: false })
+    );
```

Comment:
Open to other ideas here on how to approach usage of this `WebsocketTransport`. I've opted to go with the `setDefaultTransport` swap here to avoid having to instantiate another `SceneTreeAPIClient` with the transport and change the signature of this controller's constructor. This is also likely an issue that goes away once we can pull the public protos directly and generate a client that handles this transport changing

Reply:
This implementation isn't ideal because it's configuring the transport globally. I can't confirm this, but conceptually it could mean changing the transport of another library that's using grpc-web.

I would look into configuring the transport using the `RpcOptions` that you pass when constructing a `SceneTreeAPIClient`. On it there's a `transport` property where you can define the transport based on the RPC method you're trying to call.

Reply:
@danschultz Yeah, that's correct - the other option here would be to create a `subscriptionClient` (or similar) that is instantiated as:
```typescript
new SceneTreeAPIClient(sceneTreeHost, {
  transport: grpc.WebSocketTransport(),
});
```
I can certainly make that change, but it would mean maintaining two separate clients + a breaking change to the controller's constructor signature. I can definitely make that change though